main.tsx

```diff
@@ -143,9 +225,76 @@ class AvaSettingTab extends PluginSettingTab {
   display(): void {
     const root = createRoot(this.containerEl);
     root.render(
-        <ThemeProvider theme={theme}>
-          <CustomSettings plugin={this.plugin} />
-        </ThemeProvider>
+      <ThemeProvider theme={theme}>
+        <CustomSettings plugin={this.plugin} />
+      </ThemeProvider>
+    );
+  }
+}
+
+export class AvaSidebarView extends ItemView {
+  private readonly plugin: AvaPlugin;
+
+  constructor(leaf: WorkspaceLeaf, plugin: AvaPlugin) {
+    super(leaf);
+    this.plugin = plugin;
+  }
+
+  getDisplayText(): string {
+    return 'Wikpedia Links';
+  }
+
+  getViewType(): string {
+    return VIEW_TYPE_AVA;
+  }
+  removeLoading = () => {
+    document.getElementById('loading-state')?.remove();
+  };
+
+  setLoading = () => {
+    const loadingEl = document.createElement('div');
+    loadingEl.innerText = 'Loading...';
+    loadingEl.id = 'loading-state';
+    this.contentEl.appendChild(loadingEl);
+  };
+
+  async updateContent(content: string): Promise<void> {
+    const linkified = linkifyHtml(content);
+    await MarkdownRenderer.renderMarkdown(
+      linkified,
+      this.contentEl,
+      '',
+      this.plugin
     );
+    // this.containerEl.append(linkified);
+  }
+
+  getIcon(): string {
+    return 'clock';
+  }
+
+  async onOpen(): Promise<void> {
+    const header = document.createElement('h2');
```

Comment:
Can we only use React for consistency?

Reply:
as mentioned, won't add react rn too much overhead based on our use case (adds extra complication such as passing down state from outside a react app)